File: functions/Visualization_functions/F_Visualize_Ex1.py
import numpy as np
import matplotlib.pyplot as plt
import os

def F_Visualize_Ex1(P, S, S0, Q, K):
    lw = 2
    msize = 15
    fsize = 20
    opt_save = 1

    # Ensure figures directory exists
    if opt_save == 1 and not os.path.exists("figures"):
        os.makedirs("figures")

    # precipitation
    plt.figure()
    plt.gca().tick_params(labelsize=fsize)
    plt.bar(np.arange(1, len(P) + 1), P, color='k')
    plt.xlabel('time')
    plt.xlim([-0.5, len(P) + 0.5])
    plt.xticks(np.arange(0, 51, 10))
    plt.ylabel('P-E')
    plt.title('Input data: precipitation - evapotranspiration')
    plt.box(True)
    if opt_save == 1:
        plt.savefig('figures/Ex1_P.jpeg', dpi=300)
    plt.show()
    plt.close()

    # storage
    plt.figure()
    plt.gca().tick_params(labelsize=fsize)
    plt.plot(np.arange(0, len(P) + 1), np.insert(S, 0, S0), '.-k', linewidth=lw, markersize=msize)
    plt.xlabel('time')
    plt.xlim([-0.5, len(P) + 0.5])
    plt.xticks(np.arange(0, 51, 10))
    plt.ylabel('S')
    plt.ylim([0, np.max(np.insert(S, 0, S0)) + 2])
    plt.title('Water storage estimates - single model simulation')
    plt.box(True)
    if opt_save == 1:
        plt.savefig('figures/Ex1_S.jpeg', dpi=300)
    plt.show()
    plt.close()

    # runoff/discharge
    plt.figure()
    plt.gca().tick_params(labelsize=fsize)
    plt.plot(np.arange(1, len(P) + 1), Q, '.-k', linewidth=lw, markersize=msize)
    plt.xlabel('time')
    plt.xlim([-0.5, len(P) + 0.5])
    plt.xticks(np.arange(0, 51, 10))
    plt.ylabel('R')
    plt.ylim([0, np.max(np.insert(S, 0, S0)) + 2])
    plt.title('River discharge estimates - single model simulation')
    plt.box(True)
    if opt_save == 1:
        plt.savefig('figures/Ex1_R.jpeg', dpi=300)
    plt.show()
    plt.close()

    # relation between S and Q
    S_sim = np.arange(0, 25.1, 0.1)
    Q_sim = K * S_sim
    plt.figure()
    plt.gca().tick_params(labelsize=fsize)
    plt.plot(S_sim, Q_sim, '-k', linewidth=lw, markersize=msize)
    plt.xlabel('S')
    plt.ylabel('R')
    plt.title('Relation between storage and discharge - single model simulation')
    plt.box(True)
    plt.axis('equal')
    plt.xlim([0, 25])
    plt.ylim([0, 8])
    if opt_save == 1:
        plt.savefig('figures/Ex1_StoR.jpeg', dpi=300)
    plt.show()
    plt.close()

File: functions/Visualization_functions/F_Visualize_Ex1.m
function F_Visualize_Ex1(P,S,S0,Q,K)

lw = 2;
msize = 15;
fsize = 20;
opt_save = 1;

% precipitation
figure
set(gca,'FontSize',fsize)
bar(1:length(P),P,'k');
xlabel('time'); xlim([0-0.5 length(P)+0.5]);
set(gca, 'XTick', (0:10:50));
ylabel('P-E')
title('Input data: precipitation - evapotranspiration');
box on
if(opt_save==1) print(gcf,'figures/Ex1_P','-djpeg'); end

% storage
figure
set(gca,'FontSize',fsize)
plot(0:length(P),[S0; S],'.-k','LineWidth',lw,'MarkerSize',msize);
xlabel('time'); xlim([0-0.5 length(P)+0.5]);
set(gca, 'XTick', (0:10:50));
ylabel('S'); ylim([0 max([S0; S])+2]);
title('Water storage estimates - single model simulation');
box on
if(opt_save==1) print(gcf,'figures/Ex1_S','-djpeg'); end

% runoff/discharge
figure
set(gca,'FontSize',fsize)
plot(1:length(P),Q,'.-k','LineWidth',lw,'MarkerSize',msize);
xlabel('time'); xlim([0-0.5 length(P)+0.5]);
set(gca, 'XTick', (0:10:50));
ylabel('R'); ylim([0 max([S0; S])+2]);
title('River discharge estimates - single model simulation');
box on
if(opt_save==1) print(gcf,'figures/Ex1_R','-djpeg'); end

% relation between S and Q
S_sim = (0:0.1:25)';
Q_sim = K*S_sim;
figure
set(gca,'FontSize',fsize)
plot(S_sim,Q_sim,'-k','LineWidth',lw,'MarkerSize',msize);
xlabel('S'); ylabel('R');
title('Relation between storage and discharge - single model simulation');
box on
axis equal
xlim([0 25]); ylim([0 8]);
if(opt_save==1) print(gcf,'figures/Ex1_StoR','-djpeg'); end

end

